### DISCUSSION TITLE
How does the integration with Rainlab.Pages work?

### DISCUSSION THREAD

**Discussion by chocolata** [OFFLINE-GmbH/oc-boxes-support#159] [Help] [answered]
Hi, 

Loving the plugin. I have been using Rainlab.Pages for all of my projects... 

I noticed you mention that the plugin integrates perfectly with Rainlab Pages:
![image](https://github.com/user-attachments/assets/42bde374-ae8d-4a78-ba6e-2e7b91a403a4)

If my website already is populated with static pages, could you please elaborate how we can get the Boxes plugin working inside pages?
I don't seem to find references to that in your documentation.

Could you kindly elaborate?

**Comment by tobias-kuendig** [comment]
The RainLab pages integration means that you can reference Boxes pages in your static pages menu (or your Sitemap). 

Please not that this info pre-dates the new pagefinder widget that is now built into October, as well as the new recommended "Tailor"  Sitemap approach.

**Reply by chocolata** [reply]
Hi @tobias-kuendig 

Thank you. I was under the impression we could get the editor to work inside the Rainlab.Pages environment, but I understand this is not the case.

If I understand you correctly, we would still need to use the Rainlab.Pages plugin to create navigation menus for a site and that navigation menus are not part of the scope of the boxes plugin?

It might be confusing to users to see the ability to create pages both via the OFFLINE.Boxes environment and the Rainlab.Pages environment? Or might I be missing something?

**Reply by tobias-kuendig** [reply, marked as answer]
> If I understand you correctly, we would still need to use the Rainlab.Pages plugin to create navigation menus 

That is correct, if you still want to use the static pages plugin. Another option is to build a tailor blueprint that allows the user to generate the menu. Using the pagefinder widget to build the menu in Tailor and a Tailor globals blueprint sometimes is enough to completely replace the additional functionality the static pages plugin offers.

> It might be confusing to users to see the ability to create pages both via the OFFLINE.Boxes environment and the Rainlab.Pages environment? Or might I be missing something?

Yes, there is definitely overlap between the two plugins, so displaying them both to the end-user is probably not ideal.

**Reply by chocolata** [reply]
Thank you @tobias-kuendig , this is clear. Maybe I can do something to hide the pages section of the Rainlab.Pages plugin in my own custom plugin. I'll look into it.

**Reply by chocolata** [reply]
Hi @tobias-kuendig I wanted to return to this topic, because I had some insights. Do you think it would be feasible to sync the pages of Rainlab.Pages with the pages created in the Boxes plugin? I've centered a lot of functionality around the Rainlab.Pages plugin, and it would be great if we would have a feature like: edit this page in the Boxes pagebuilder. A bit like how the pagebuilders in Wordpress (I know...) work... You have the option to use the Wordpress editing tools, and you have the ability to edit the page in a visual page builder.

**Reply by tobias-kuendig** [reply]
The Boxes Editor can be used in plugins: https://docs.boxes.offline.ch/use-cases/usage-in-plugins.html

So in theory, it can be integrated in RainLab.Pages (replacing the existing markup editor). But I have to be honest: Boxes has replaced the RainLab.Pages plugin for us, so we don't have any use for the integration and probably are not going to spend time developing it.

Since October is extremely extendible, you should be able to get the editor working without having to modify either Boxes or the Pages plugin by creating a custom "glue" plugin that replaces the editor in Pages. But keep in mind that all this is just theoretical and the devil is, as always, in the detail.

**Comment by chocolata** [comment]
Thank you Tobias, for your clear answer. I appreciate it.
